platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/recovery/internal/EventStreamPathIterator.java

```diff
@@ -59,10 +66,26 @@ public EventStreamPathIterator(final Path eventStreamDirectory, final long start
                     .forEachOrdered(eventStreamFiles::add);
         }
 
-        if (startingRound <= FIRST_ROUND_AVAILABLE) {
-            // We are attempting to get events from the beginning of time.
+        if (eventStreamFiles.isEmpty()) {
+            iterator = Collections.emptyIterator();
+            return;
+        }
+
+        final DetailedConsensusEvent firstEvent = getFirstEventInEventStreamFile(eventStreamFiles.get(0));
+        if (bound.isUnbounded() || bound.compareTo(firstEvent, BoundType.LOWER) == 0) {
+            // We are attempting to get events from the beginning of the event stream.
             iterator = eventStreamFiles.iterator();
         } else {
+            final EventStreamBound usedBound;
+            if (bound.hasRound()) {
+                // corner case: get the file containing the round prior since the file may also contain events
```

Comment:
I'm having trouble wrapping my head around *why* this corner case exists. Could this explanation be expanded a bit to make it more clear what's happening here?

Reply:
I don't think this is a corner case, it's the standard case if we say "start at round X".

The reason why we subtract 1 from the bound is because we only read the first event from each file when we are doing the binary search. This means that if we find a file that starts with an event from the correct round, we can't be guaranteed that the previous file doesn't have events at the end with the round we need. So we keep searching until we find a file with a previous round, which guarantees that the events we need are in this file or later files.

It might be useful to add a comment saying as much for future readers of this code.

Reply:
@cody-littley Is it guaranteed that the events of a single round will not span more than 2 stream files?

Reply:
updated wording.